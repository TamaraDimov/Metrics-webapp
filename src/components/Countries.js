import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCountries } from '../redux/countrySlice';
import Country from './Country';
import worldMap from '../img/worldM.png';

const Countries = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries);

  const [continent, setContinent] = useState('all');

  useEffect(() => {
    if (countries.length === 0) {
      dispatch(fetchCountries());
    }
  }, [countries, dispatch]);

  return (
    <div className="main">
      <img src={worldMap} alt="worls map" className="logo" />
      <h1 className="title">All Countries</h1>
      <div className="dropdownMenu">
        <select
          name="continents"
          id="continents"
          className="select"
          onChange={(e) => setContinent(e.target.value)}
        >
          <option value="all">All</option>
          <option value="Antarctica">Antarctica</option>
          <option value="Africa">Africa</option>
          <option value="Asia">Asia</option>
          <option value="Europe">Europe</option>
          <option value="Oceania">Oceania</option>
          <option value="South America">South America</option>
          <option value="North America">North America</option>
        </select>
      </div>
      <ul className="countries">
        {countries
          .filter((country) => {
            if (continent === 'all') {
              return country;
            }
            return country.continent === continent;
          })
          .map((country) => (
            <Country
              key={country.name}
              name={country.name}
              continent={country.continent}
              flagEmoji={country.flagEmoji}
            />
          ))}
      </ul>
    </div>
  );
};

export default Countries;
