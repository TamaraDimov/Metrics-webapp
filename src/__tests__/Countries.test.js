import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Countries from '../components/Countries';

describe('Countries', () => {
  it('renders component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Countries />
        </BrowserRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
