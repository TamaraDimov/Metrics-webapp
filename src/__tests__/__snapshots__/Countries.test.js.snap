// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries renders component correctly 1`] = `
<div>
  <div
    class="main"
  >
    <img
      alt="worls map"
      class="logo"
      src="worldM.png"
    />
    <h1
      class="title"
    >
      All Countries
    </h1>
    <div
      class="dropdownMenu"
    >
      <select
        class="select"
        id="continents"
        name="continents"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="Antarctica"
        >
          Antarctica
        </option>
        <option
          value="Africa"
        >
          Africa
        </option>
        <option
          value="Asia"
        >
          Asia
        </option>
        <option
          value="Europe"
        >
          Europe
        </option>
        <option
          value="Oceania"
        >
          Oceania
        </option>
        <option
          value="South America"
        >
          South America
        </option>
        <option
          value="North America"
        >
          North America
        </option>
      </select>
    </div>
    <ul
      class="countries"
    />
  </div>
</div>
`;
